### PR TITLE
Added support for `databricks_mws_workspaces` on GCP (Public Preview)

### PR DESCRIPTION
**Pull request by nkvuong**
Docs for #1871 

- Update `databricks_mws_networks` and `databricks_mws_workspaces` doc
- Rename `AWS` subcategory to `Deployment`
- Add GCP auth information to index

### CODE REVIEW

File: docs/resources/mws_networks_gcp.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
+---
+subcategory: "GCP"
```
Comment (nfx) [inline docs/resources/mws_networks_gcp.md:2]: this creates a strange precedent. do we want it?

Comment (nkvuong) [reply docs/resources/mws_networks_gcp.md:2]: `databricks_mws_workspaces` and `databricks_mws_networks` would apply to both GCP & AWS, so we will need a new subcategory for both resources?

**Comment by grayluck** [comment]
Expected change log: Add Terrafrom schema support for GCP account API public preview workspace management.

**Review by nfx — CHANGES_REQUESTED** [review]
change suffices

**Pull request merged**